arcaflow_plugin_stressng/stressng_schema.py

```diff
@@ -33,104 +190,356 @@ class CommonStressorParams:
 
 @dataclass
 class CpuStressorParams(CommonStressorParams):
-    cpu_method: typing.Annotated[
-        typing.Optional[str],
-        schema.name("CPU Stressor Method"),
-        schema.description(
-            "Fine grained control of which "
-            "CPU stressors to use (ackermann, "
-            "cfloat etc.)"
-        ),
-    ] = "all"
+    cpu_ops: typing.Annotated[
+        typing.Optional[int],
+        schema.id("cpu-ops"),
+        schema.name("CPU Operations"),
+        schema.description("Stop CPU stress workers after N bogo operations"),
+    ] = None
 
     cpu_load: typing.Annotated[
         typing.Optional[int],
+        schema.id("cpu-load"),
         schema.name("CPU Load"),
-        schema.description("Load CPU by percentage"),
+        schema.description(
+            "Load CPU with P percent loading for the CPU stress workers"
+        ),
     ] = None
 
+    cpu_method: typing.Annotated[
+        typing.Optional[CpuMethod],
+        schema.id("cpu-method"),
+        schema.name("CPU Stressor Method"),
+        schema.description(
+            "Specify a cpu stress method; by default, all stress methods "
+            "are exercised sequentially"
+        ),
+    ] = CpuMethod.ALL
```

Comment:
Unless there is some way for the workflow to supply a `None` or `nil` value, line 210 should omit the `Optional`, since the default value is a `CpuMethod`.

Reply:
I'm not sure of another example in a plugin where we assign a default value to a non-optional parameter. It seems like this is convention, though I'm not sure why. I do know that the SDK enforces any optional parameter to have a default value assigned, though that is usually `None`.

Reply:
In this context, `Optional` applies to the _type_ not the _value_.  🤪

In the other cases, where we assign `None` to an attribute which is supposed to be (e.g.) an `int` (like `cpu_ops`), we have to make the "`int`-ness" _optional_, because `None` is not an `int` type -- that is, if the type weren't optional, we couldn't (legitimately) set it to `None`.

Whether there is a default initializer is always optional (although, if you omit it, what you get is pretty context-sensitive).

(Of course, this is all assuming _typical_ Python usage...I'm not sure whether the Arca magic changes these rules....)

Reply:
I don't think that's accurate. The SDK uses `Typing.Optional` to indicate whether the input parameter is required or optional for the plugin. In this case, the user is not required to provide a `cpu_method` in the plugin input, and if they do not, then the default value from the enum of `all` will be applied, tranparent to the user.

Reply:
Oh, good...so, we took an existing meaning and redefined it.  😬 

I suppose, if we start from a position of strong typing, then `Optional` has no other meaning, so we might as well reapply it....  🥴